src/stdune/path.mli

```diff
@@ -87,6 +109,10 @@ module Build : sig
   (** set the build directory. Can only be called once and must be done before
       paths are converted to strings elsewhere. *)
   val set_build_dir : Kind.t -> unit
+
+  val of_relative : Relative.t -> t
```

Comment:
What do we do about attempts to go outside the root here?

Reply:
There should be an error. I'm thinking that this should be equivalent to `Local.of_relative relative |> Path.Build.of_local`

Reply:
I'd expect this to be indicated in the type somehow. (`_exn`?). (if I'm OK with `of_string` raising, it was mostly because of_string conventionally raises; conversions between our own types I'd expect to be explicit about errors)

Reply:
Yeah, I agree with that. But I think we should start with `Path.relative`, `Path.Build.relative`, etc. All of those can raise at the moment.

Reply:
Fair enough, I don't insist.

Reply:
I'll leave this to a later PR.